### PR TITLE
docs: add dynamic as signer

### PR DESCRIPTION
**Pull request by matthew1809**

### CODE REVIEW

File: site/snippets/dynamic.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { WalletClientSigner, type SmartAccountSigner } from "@alchemy/aa-core";
+import { useDynamicContext } from "@dynamic-labs/sdk-react-core";
+
+const { primaryWallet } = useDynamicContext();
+
+const dynamicProvider = await primaryWallet?.connector?.getWeb3Provider();
+
+// a smart account signer you can use as an owner on ISmartContractAccount
+export const dynamicSigner: SmartAccountSigner = new WalletClientSigner(
+  dynamicProvider
```
Comment (moldy530) [inline site/snippets/dynamic.ts:10]: we can definitely improve this and we will, but right now `WalletClientSigner` requires a `viem` `WalletClient`.

To do that you would have to do something like this:
```suggestion
  createWalletClient({ chain: getChain(dynamicProvider.chainId), transport: custom(dynamicProvider) })
```

`getChain` is a method we export as a util from aa-core, so you'll have to import that as well (and then pass in a chainId however you get it from the dynamicProvider)
`custom` and `createWalletClient` are `viem` imports.

Comment (matthew1809) [reply site/snippets/dynamic.ts]: ack, thanks!

File: site/snippets/dynamic.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { WalletClientSigner, type SmartAccountSigner } from "@alchemy/aa-core";
+import { useDynamicContext } from "@dynamic-labs/sdk-react-core";
+
+const { primaryWallet } = useDynamicContext();
+
+const dynamicProvider = await primaryWallet?.connector?.getWeb3Provider();
+
+// a smart account signer you can use as an owner on ISmartContractAccount
+export const dynamicSigner: SmartAccountSigner = new WalletClientSigner(
+  dynamicProvider,
```
Comment (moldy530) [inline site/snippets/dynamic.ts:10]: this still needs to be wrapped in `createWalletClient`

```suggestion
  createWalletClient({ 
    transport: custom(dynamicProvider),
    chain: getChain(network!),
  }),
```

File: site/snippets/dynamic.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { WalletClientSigner, type SmartAccountSigner } from "@alchemy/aa-core";
+import { useDynamicContext } from "@dynamic-labs/sdk-react-core";
+
+const { primaryWallet } = useDynamicContext();
```
Comment (moldy530) [inline site/snippets/dynamic.ts:4]:
```suggestion
const { primaryWallet, network } = useDynamicContext();
```

File: site/snippets/dynamic.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { WalletClientSigner, type SmartAccountSigner } from "@alchemy/aa-core";
+import { useDynamicContext } from "@dynamic-labs/sdk-react-core";
```
Comment (moldy530) [inline site/snippets/dynamic.ts:2]:
```suggestion
import { useDynamicContext } from "@dynamic-labs/sdk-react-core";
import { createWalletClient, custom } from "viem";
import { getChain } from "@alchemy/aa-core";
```

File: site/snippets/dynamic.ts (hunk 2 above)
Comment (matthew1809) [reply site/snippets/dynamic.ts:12]: So with Dynamic (latest), getWeb3Provider already returns a Viem WalletClient i.e. the viem interface to interact with the wallet @moldy530

Comment (moldy530) [reply site/snippets/dynamic.ts:12]: OHHH ok I missed that, that's my bad

**Review by moldy530 — APPROVED** [review]

File: site/snippets/dynamic.ts (hunk 2 above)
Comment (matthew1809) [reply site/snippets/dynamic.ts:12]: In fact this is a little misleading, we will get the method renamed to getWalletClient, let me update this

**Review by moldy530 — APPROVED** [review]
LGTM, might need to rebase though

**Pull request closed without merging**